ha/core/system_health/system_health.py

```diff
@@ -292,11 +292,12 @@ def _is_update_required(self, current_health: str, updated_health: str, event: H
             event_action = HEALTH_EVENT_ACTIONS.IGNORE.value
 
         # specific cases
-        # 1. Do not overwrite node status to failed if offline already
+        # 1. Do not overwrite node status to offline if failed already
         if event.resource_type == RESOURCE_TYPES.NODE.value and \
-            old_status == HEALTH_STATUSES.OFFLINE.value and new_status == HEALTH_STATUSES.FAILED.value:
+            old_status == HEALTH_STATUSES.FAILED.value and new_status == HEALTH_STATUSES.OFFLINE.value:
```

Comment:
As discussed this needs to handle other cases like recovering , recovered etc also

Reply:
As listed in LLD Possible values for health status for “node” resource: "online" | "degraded" | "offline" | "failed" | "recovering"
Added check for  "degraded" | "offline" | "recovering"

Test Logs :
2022-04-18 10:33:54 fault_tolerance [7]: INFO [_is_update_required] Updating is not needed node is in failed state and received offline state
2022-04-18 10:34:52 fault_tolerance [7]: INFO [_is_update_required] Updating is not needed node is in failed state and received recovering state
For online event from k8s monitor : event sent to hare